### PR TITLE
fix(siglip2): clamp logit_scale to avoid overflow

### PR DESCRIPTION
**Pull request by prabhasg5**
## 📝 Description

**⚠️ Issue Link Required**: This PR must be linked to an approved and assigned issue. See [Contributing Guide](https://github.com/kornia/kornia/blob/main/CONTRIBUTING.md#pull-request) for details.

**Fixes/Relates to:** #3481

**Important**:
- Ensure you are assigned to the linked issue before submitting this PR
- This PR should strictly implement what the linked issue describes
- Do not include changes beyond the scope of the linked issue

---

## 🛠️ Changes Made
- [x] Clamp `logit_scale` parameter before `exp()` in `SigLip2Model.forward()` to prevent numerical overflow
- [x] Added `math` import for `log(100.0)` constant (maximum scale cap)
- [x] Added unit test `test_logit_scale_clamping()` to verify clamping behavior

**Rationale:**
- Prevents unbounded growth of temperature parameter during training
- Maintains numerical stability in logit computation (max effective scale: 100)
- Follows established CLIP practice for vision-language models
- Zero behavioral impact for normal training ranges

---

## 🧪 How Was This Tested?
- [x] **Unit Tests:**
  - Added `test_logit_scale_clamping()` - verifies that oversized logit_scale values are clamped and forward pass produces finite logits
  - All 11 existing `TestSigLip2Model` tests pass (smoke, cardinality, exception, gradient, dynamo, etc.)
  
- [x] **Manual Verification:**
  - Set `logit_scale` to large value (100.0) and verified no NaN/inf in output
  - Confirmed `logits_per_image` and `logits_per_text` remain finite
  - Verified effective scale does not exceed `exp(log(100)) = 100`

- [x] **Edge Cases:**
  - Tested with oversized logit_scale values (overflow case)
  - Tested with normal initialization (no behavior change)
  - Tested on CPU with float32 dtype

---

## 🕵️ AI Usage Disclosure
- [x] 🟡 **AI-assisted:** Used AI for initial implementation structure and test pattern, but manually reviewed every line, verified correct clamping logic, and tested all code locally.

---

## 🚦 Checklist
- [ ] I am assigned to the linked issue (required before PR submission)
- [ ] The linked issue has been approved by a maintainer
- [x] I have performed a **self-review** of my code (clamping value chosen from CLIP standards, minimal change scope)
- [x] My code follows the existing style guidelines of this project (PEP8, type hints, docstrings match existing patterns)
- [x] I have added tests that prove my fix is effective (test_logit_scale_clamping validates clamping)
- [x] All local tests pass: `pixi run test tests/models/test_siglip2.py::TestSigLip2Model`

---

## 💭 Additional Context
**Why this change:**
The `logit_scale` parameter (temperature) in contrastive vision-language models can grow unbounded during training if unchecked. Clamping prevents:
- `exp()` overflow → NaN propagation through loss
- Numerical instability in similarity computations
- Silent training failures

**Impact:** 
- Single-line fix with comprehensive test coverage
- No API changes, fully backward compatible
- Minimal performance overhead (one `clamp()` call per forward pass)

**Related work:** CLIP and other vision-language models use similar clamping (e.g., log_scale_learnable <= log(100) in OpenAI CLIP).

### CODE REVIEW

File: kornia/models/siglip2/model.py [hunk 1]
```diff
@@ -32,6 +33,9 @@
 
 __all__ = ["SigLip2Model", "SigLip2Result"]
 
+# Maximum value for logit_scale to prevent overflow: exp(log(100)) = 100
+LOGIT_SCALE_MAX = 100.0
```
Comment (edgarriba) [inline kornia/models/siglip2/model.py]: maybe include it in the siglip config ?

File: tests/models/test_siglip2.py [hunk 2]
```diff
@@ -193,6 +193,61 @@ def test_dynamo(self, device, dtype, torch_optimizer, model, config):
         self.assert_close(actual.image_embeds, expected.image_embeds)
         self.assert_close(actual.text_embeds, expected.text_embeds)
 
+    def test_logit_scale_clamping(self, device, dtype, config):
```
Comment (edgarriba) [inline tests/models/test_siglip2.py]: too much to add two test for this. think how to adapt slightly other tests to see that change reflected

File: kornia/models/siglip2/model.py (hunk 1 above)
Comment (prabhasg5) [reply kornia/models/siglip2/model.py]: yes! may be that makes more sense and avoids confusion. i'll change

File: tests/models/test_siglip2.py (hunk 2 above)
Comment (prabhasg5) [reply tests/models/test_siglip2.py]: so shall we consolidate the clamping tests into one and integrate it with existing test logic ?

**Comment by prabhasg5** [comment]
Both addressed: moved to config + consolidated tests into test_return_loss. All passing.

File: kornia/models/siglip2/model.py [hunk 3]
```diff
@@ -217,7 +218,7 @@ def forward(
             else None
         )
 
-        logit_scale = self.logit_scale.exp()
+        logit_scale = self.logit_scale.clamp(min=0.0, max=math.log(self.config.logit_scale_max)).exp()
```
Comment (edgarriba) [inline kornia/models/siglip2/model.py]: also i believe this can be cached in the class as it's a constant value

Comment (prabhasg5) [reply kornia/models/siglip2/model.py]: Good point, agreed. I will cache the logit_scale_max_log value in the module and use it here instead of recomputing it each time. I will push and update now

**Comment by prabhasg5** [comment]
and also shall I type hint for cached value like this

self.logit_scale_max_log: float = math.log(config.logit_scale_max) ?

**Comment by prabhasg5** [comment]
All checks are now passing and commits are signed.
Happy to help with anything else if needed.

**Review by edgarriba — APPROVED** [review]
